packages/react-instantsearch-nextjs/src/useNextHeaders.tsx

```diff
@@ -0,0 +1,28 @@
+import { headers } from 'next/headers';
+import { use } from 'react';
+
+function isPromise(obj: any) {
+  return (
+    obj &&
+    (typeof obj === 'object' || typeof obj === 'function') &&
+    typeof obj.then === 'function'
+  );
+}
+
+type Headers = Awaited<ReturnType<typeof headers>> | undefined;
```

Comment:
Nitpick: should we remove the union with `undefined`, throwing instead if `useNextHeaders()` is called from a non-server environment?

We are confident we are calling it properly and it will remove the need for optional chaining in its usage.

Reply:
We don't want it to throw as it has to be called on the frontend for hydration no ?